### PR TITLE
[PE-5022] Updated pagination to support data table use cases

### PR DESCRIPTION
**Pull request by mattnickles**
Updated pagination component to support data table use cases:

**Page Size:**
Drop down to allow users to configure the number of result items per page.

**Page Jumper:**
Input to allow users to jump to a specific page of results.

**Results Label:**
Label to indicate the total number of results.

**Truncation:**
When the number of pages exceeds the recommended page limit, an ellipsis is used to truncate the remaining pages.

**Double Truncation:**
Double truncation is used when the current page is separated by 5 or more pages from the first and last page.

**Compact:**
Miniature version of the component for when space is limited.

**First and Last Page Buttons:**
Adds two extra icon buttons which allow navigating to the first and last page.

### CODE REVIEW

**Comment by allyourbotsarebelongtous** [comment]
Your CHI Pull Request website is available for review at <a href="https://ee4692ef759d60e72e819bcef45860c1b120b790-chi-pr.enterprise.ctl.io" target="_blank"> https://ee4692ef759d60e72e819bcef45860c1b120b790-chi-pr.enterprise.ctl.io </a>

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/1001/ for more details.

**Comment by allyourbotsarebelongtous** [comment]
Your CHI Pull Request website is available for review at <a href="https://742ab894c95d363411f026313427bb4dd383ca4f-chi-pr.enterprise.ctl.io" target="_blank"> https://742ab894c95d363411f026313427bb4dd383ca4f-chi-pr.enterprise.ctl.io </a>

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/1002/ for more details.

File: src/website/views/components/pagination/_examples.pug [hunk 1]
```diff
@@ -7,39 +8,1008 @@ h3 Base
...
+        a(href=`#webcomponent-results-label`) Web Component
+      li.-active
+        a(href=`#html-results-label`) HTML Blueprint
+  .chi-tabs-panel#webcomponent-results-label
+  .chi-tabs-panel.-active#html-results-label
+    :code(lang="html")
+      <!-- For light backgrounds -->
+      <nav class="chi-pagination" role="navigation" aria-label="Pagination">
+        <div class="chi-pagination__content">
+          <div class="chi-pagination__start">
```
Comment (jllr) [inline src/website/views/components/pagination/_examples.pug:520]: @mattnickles please add closing div tag

File: src/website/views/components/pagination/_examples.pug [hunk 2]
```diff
@@ -7,39 +8,1008 @@ h3 Base
...
+      span.-text--grey
+        | This HTML Blueprint requires JavaScript.
+        | You may use your own solution, or use Chi's vanilla JavaScript solution,
+        | <a href="../../getting-started/installation">Chi.js</a>.
+    :code(lang="html")
+      <!-- For light backgrounds -->
+      <!-- HTML -->
+      <nav class="chi-pagination" role="navigation" aria-label="Pagination">
+        <div class="chi-pagination__content">
+          <div class="chi-pagination__start">
```
Comment (jllr) [inline src/website/views/components/pagination/_examples.pug:675]: @mattnickles please add closing div tag

File: src/website/views/components/pagination/_examples.pug [hunk 3]
```diff
@@ -7,39 +8,1008 @@ h3 Base
...
+        </div>
+      </nav>
+      <!-- JavaScript -->
+      <script>chi.dropdown(document.getElementById('example-pagesize-1'));</script>
+
+      <!-- For dark backgrounds -->
+      <!-- HTML -->
+      <nav class="chi-pagination -inverse" role="navigation" aria-label="Pagination">
+        <div class="chi-pagination__content">
+          <div class="chi-pagination__start">
```
Comment (jllr) [inline src/website/views/components/pagination/_examples.pug:724]: @mattnickles please add closing div tag

File: src/website/views/components/pagination/_examples.pug [hunk 4]
```diff
@@ -50,131 +1020,314 @@ h3 Base
...
-          <a href="#" class="chi-button -flat">4</a>
-          <a href="#" class="chi-button -flat">5</a>
-          <a href="#" class="chi-button -icon -flat" aria-label="Next">
-            <div class="chi-buton__content">
-              <i class="chi-icon icon-chevron-right" aria-hidden="true"></i>
-            </div>
-          </a>
+      <nav class="chi-pagination -compact" role="navigation" aria-label="Pagination">
+        <div class="chi-pagination__content">
+          <div class="chi-pagination__start">
```
Comment (jllr) [inline src/website/views/components/pagination/_examples.pug:1025]: @mattnickles please add closing div tag

File: src/website/views/components/pagination/_examples.pug [hunk 5]
```diff
@@ -50,131 +1020,314 @@ h3 Base
...
-          <a href="#" class="chi-button -flat -light">4</a>
-          <a href="#" class="chi-button -flat -light">5</a>
-          <a href="#" class="chi-button -icon -flat -light" aria-label="Next">
-            <div class="chi-button__content">
-              <i class="chi-icon icon-chevron-right" aria-hidden="true"></i>
-            </div>
-          </a>
+      <nav class="chi-pagination -compact -inverse" role="navigation" aria-label="Pagination">
+       <div class="chi-pagination__content">
+          <div class="chi-pagination__start">
```
Comment (jllr) [inline src/website/views/components/pagination/_examples.pug:1054]: @mattnickles please add closing div tag

File: src/website/views/components/pagination/_examples.pug [hunk 6]
```diff
@@ -50,131 +1020,314 @@ h3 Base
...
-          <a href="#" class="chi-button -flat">4</a>
-          <a href="#" class="chi-button -flat">5</a>
-          <a href="#" class="chi-button -icon -flat" aria-label="Next">
-            <div class="chi-button__content">
-              <i class="chi-icon icon-chevron-right" aria-hidden="true"></i>
-            </div>
-          </a>
+      <nav class="chi-pagination -compact" role="navigation" aria-label="Pagination">
+        <div class="chi-pagination__content">
+          <div class="chi-pagination__start">
```
Comment (jllr) [inline src/website/views/components/pagination/_examples.pug:1135]: @mattnickles please add closing div tag

File: src/website/views/components/pagination/_examples.pug [hunk 7]
```diff
@@ -50,131 +1020,314 @@ h3 Base
...
-          <a href="#" class="chi-button -flat -light">4</a>
-          <a href="#" class="chi-button -flat -light">5</a>
-          <a href="#" class="chi-button -icon -flat -light" aria-label="Next">
-            <div class="chi-button__content">
-              <i class="chi-icon icon-chevron-right" aria-hidden="true"></i>
-            </div>
-          </a>
+      <nav class="chi-pagination -compact -inverse" role="navigation" aria-label="Pagination">
+       <div class="chi-pagination__content">
+          <div class="chi-pagination__start">
```
Comment (jllr) [inline src/website/views/components/pagination/_examples.pug:1166]: @mattnickles please add closing div tag

File: src/website/views/components/pagination/_examples.pug [hunk 8]
```diff
@@ -50,131 +1020,314 @@ h3 Base
...
+        a(href=`#webcomponent-base`) Web Component
+      li.-active
+        a(href=`#html-base`) HTML Blueprint
+  .chi-tabs-panel#webcomponent-base
+  .chi-tabs-panel.-active#html-base
+    :code(lang="html")
+      <!-- For light backgrounds -->
+      <nav class="chi-pagination -compact" role="navigation" aria-label="Pagination">
+        <div class="chi-pagination__content">
+          <div class="chi-pagination__start">
```
Comment (jllr) [inline src/website/views/components/pagination/_examples.pug:1259]: @mattnickles please add closing div tag

File: src/website/views/components/pagination/_examples.pug [hunk 9]
```diff
@@ -50,131 +1020,314 @@ h3 Base
...
+            </div>
+          </div>
+          <div class="chi-pagination__end"></div>
+        </div>
+      </nav>
+
+      <!-- For dark backgrounds -->
+      <nav class="chi-pagination -compact -inverse" role="navigation" aria-label="Pagination">
+        <div class="chi-pagination__content">
+          <div class="chi-pagination__start">
```
Comment (jllr) [inline src/website/views/components/pagination/_examples.pug:1298]: @mattnickles please add closing div tag

File: src/website/views/components/pagination/_examples.pug [hunk 10]
```diff
@@ -7,39 +8,1008 @@ h3 Base
...
+                  else if step == '6'
+                    a(class=`chi-button -icon -flat`, href='#', aria-label='Next page')
+                      .chi-button__content
+                        i(class="chi-icon icon-chevron-right", aria-hidden='true')
+                  else
+                    a(class=`chi-button -flat ${step === '3' ? `-active` : ''}`, href='#')= `${step}`
+            .chi-pagination__end
+    .chi-col.-w--12.-w-md--6
+      .-p--3.-bg--black
+        nav.chi-pagination.-inverse(role='navigation', aria-label='Pagination')
```
Comment (jllr) [inline src/website/views/components/pagination/_examples.pug:31]: @mattnickles I think -inverse class is not needed anymore. Can you remove it?

File: src/website/views/components/pagination/_examples.pug [hunk 11]
```diff
@@ -7,39 +8,1008 @@ h3 Base
...
+                </div>
+              </a>
+            </div>
+          </div>
+          <div class="chi-pagination__end"></div>
+        </div>
+      </nav>
+
+      <!-- For dark backgrounds -->
+      <nav class="chi-pagination -inverse" role="navigation" aria-label="Pagination">
```
Comment (jllr) [inline src/website/views/components/pagination/_examples.pug:85]: @mattnickles I think -inverse class is not needed anymore. Can you remove it?

File: src/website/views/components/pagination/_examples.pug [hunk 12]
```diff
@@ -7,39 +8,1008 @@ h3 Base
...
+                  else if step == '6'
+                    a(class=`chi-button -icon -flat -disabled`, href='#', aria-label='Next page')
+                      .chi-button__content
+                        i(class="chi-icon icon-chevron-right", aria-hidden='true')
+                  else
+                    a(class=`chi-button -flat -disabled ${step === '1' ? `-active` : ''}`, href='#')= `${step}`
+            .chi-pagination__end
+    .chi-col.-w--12.-w-md--6
+      .-p--3.-bg--black
+        nav.chi-pagination.-inverse(role='navigation', aria-label='Pagination')
```
Comment (jllr) [inline src/website/views/components/pagination/_examples.pug:136]: @mattnickles I think -inverse class is not needed anymore. Can you remove it?

File: src/website/views/components/pagination/_examples.pug [hunk 13]
```diff
@@ -7,39 +8,1008 @@ h3 Base
...
+                </div>
+              </a>
+            </div>
+          </div>
+          <div class="chi-pagination__end"></div>
+        </div>
+      </nav>
+
+      <!-- For dark backgrounds -->
+      <nav class="chi-pagination -inverse" role="navigation" aria-label="Pagination">
```
Comment (jllr) [inline src/website/views/components/pagination/_examples.pug:190]: @mattnickles I think -inverse class is not needed anymore. Can you remove it?

File: src/website/views/components/pagination/_examples.pug [hunk 14]
```diff
@@ -7,39 +8,1008 @@ h3 Base
...
```
Comment (jllr) [inline src/website/views/components/pagination/_examples.pug:305]: @mattnickles I think -inverse class is not needed anymore. Can you remove it?

File: src/website/views/components/pagination/_examples.pug [hunk 15]
```diff
@@ -7,39 +8,1008 @@ h3 Base
...
```
Comment (jllr) [inline src/website/views/components/pagination/_examples.pug:424]: @mattnickles I think -inverse class is not needed anymore. Can you remove it?

File: src/website/views/components/pagination/_examples.pug [hunk 16]
```diff
@@ -186,40 +1339,44 @@ p.-text Pagination supports the following sizes: <code>-sm</code>, <code>-md</co
...
-                      i(class="chi-icon icon-chevron-left", aria-hidden='true')
-                else if step == '3'
-                  a(class=`chi-button -flat -active -light -${size}`, href='#')= `${step}`
-                else if step == '6'
-                  a(class=`chi-button -icon -flat -light -${size}`, href='#', aria-label='Next')
-                    .chi-button__content
-                      i(class="chi-icon icon-chevron-right", aria-hidden='true')
-                else
-                  a(class=`chi-button -flat -light -${size}`, href='#')= `${step}`
+          nav.chi-pagination.-inverse.-mb--2(role='navigation', aria-label='Pagination')
```
Comment (jllr) [inline src/website/views/components/pagination/_examples.pug:1363]: @mattnickles I think -inverse class is not needed anymore. Can you remove it?

File: src/website/views/components/pagination/_examples.pug [hunk 17]
```diff
@@ -232,170 +1389,218 @@ p.-text Pagination supports the following sizes: <code>-sm</code>, <code>-md</co
...
+              </a>
+            </div>
+          </div>
+          <div class="chi-pagination__end"></div>
         </div>
       </nav>
 
       <!-- For dark backgrounds -->
       <!-- sm -->
       <nav class="chi-pagination -inverse" role="navigation" aria-label="Pagination">
```
Comment (jllr) [inline src/website/views/components/pagination/_examples.pug:1500]: @mattnickles I think -inverse class is not needed anymore. Can you remove it?

File: src/website/views/components/pagination/_examples.pug [hunk 18]
```diff
@@ -232,170 +1389,218 @@ p.-text Pagination supports the following sizes: <code>-sm</code>, <code>-md</co
...
+                </div>
+              </a>
+            </div>
+          </div>
+          <div class="chi-pagination__end"></div>
         </div>
       </nav>
 
       <!-- md -->
       <nav class="chi-pagination -inverse" role="navigation" aria-label="Pagination">
```
Comment (jllr) [inline src/website/views/components/pagination/_examples.pug:1527]: @mattnickles I think -inverse class is not needed anymore. Can you remove it?

File: src/website/views/components/pagination/_examples.pug [hunk 19]
```diff
@@ -232,170 +1389,218 @@ p.-text Pagination supports the following sizes: <code>-sm</code>, <code>-md</co
...
+                </div>
+              </a>
+            </div>
+          </div>
+          <div class="chi-pagination__end"></div>
         </div>
       </nav>
 
       <!-- lg -->
       <nav class="chi-pagination -inverse" role="navigation" aria-label="Pagination">
```
Comment (jllr) [inline src/website/views/components/pagination/_examples.pug:1554]: @mattnickles I think -inverse class is not needed anymore. Can you remove it?

File: src/website/views/components/pagination/_examples.pug [hunk 20]
```diff
@@ -232,170 +1389,218 @@ p.-text Pagination supports the following sizes: <code>-sm</code>, <code>-md</co
...
+                </div>
+              </a>
+            </div>
+          </div>
+          <div class="chi-pagination__end"></div>
         </div>
       </nav>
 
       <!-- xl -->
       <nav class="chi-pagination -inverse" role="navigation" aria-label="Pagination">
```
Comment (jllr) [inline src/website/views/components/pagination/_examples.pug:1581]: @mattnickles I think -inverse class is not needed anymore. Can you remove it?

**Comment by mattnickles** [comment]
@jllr The `-inverse` class is necessary for extensibility. While the first example can technically render fine without it, the styles would break if a developer adds a label, page size, or any other pagination sub component.

File: src/website/views/components/pagination/_examples.pug (hunk 1 above)
Comment (mattnickles) [reply src/website/views/components/pagination/_examples.pug:520]: Closing div tag is located on line 524

File: src/website/views/components/pagination/_examples.pug (hunk 2 above)
Comment (mattnickles) [reply src/website/views/components/pagination/_examples.pug:675]: closing div tag is located on line 692

File: src/website/views/components/pagination/_examples.pug (hunk 3 above)
Comment (mattnickles) [reply src/website/views/components/pagination/_examples.pug:724]: closing div tag is located on line 741

**Comment by allyourbotsarebelongtous** [comment]
Your CHI Pull Request website is available for review at <a href="https://673904fd811eb89cbc4eb13888b79233e2b182a8-chi-pr.enterprise.ctl.io" target="_blank"> https://673904fd811eb89cbc4eb13888b79233e2b182a8-chi-pr.enterprise.ctl.io </a>

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/1007/ for more details.

**Comment by allyourbotsarebelongtous** [comment]
The following instances are being terminated: https://ops.elasticbox.com/services/instances/i-lllz22, https://ops.elasticbox.com/services/instances/i-wyhr9n, https://ops.elasticbox.com/services/instances/i-uvtful

**Pull request merged**